src/logging.cpp

```diff
@@ -352,7 +352,7 @@ std::string BCLog::Logger::LogTimestampStr(const std::string& str)
         const auto now{SystemClock::now()};
         const auto now_seconds{std::chrono::time_point_cast<std::chrono::seconds>(now)};
         strStamped = FormatISO8601DateTime(TicksSinceEpoch<std::chrono::seconds>(now_seconds));
-        if (m_log_time_micros) {
+        if (m_log_time_micros && !strStamped.empty()) {
```

Comment:
Could wrap this in an `Assume()` to both make it clear to the reader that it's not supposed to happen and easier to catch any (very unexpected) bugs without crashing non-development builds?
```suggestion
        if (m_log_time_micros && Assume(!strStamped.empty())) {
```